File: Laba5/C#/Task2/Elipse.cs
using System.Linq;
using static System.Math;
namespace ConsoleApp1
{
    public class Elipse : Figure
    {
        public Elipse(int axis1, int axis2) : base(axis1, axis2)
        {
            this.data = new[] {axis1, axis2};
        }

        protected internal override double calculateLength() => base.calculateLength() * PI;

        protected internal override double calculateArea() => PI * data.Aggregate((x,y) => x * y);
        
        }
}

File: Laba5/C++/Task2/Elipse.cpp
#include "Elipse.h"


Elipse::Elipse(int a, int b) : Figure(new int[2] {a,b}) {
	this->data = new int[2] {a, b};
}


double Elipse::CalculateLength() {
	return (data[0] + data[1]) * M_PI;
}

double Elipse::CalculateArea() {
	return (double)(data[0] * data[1] * M_PI);
}